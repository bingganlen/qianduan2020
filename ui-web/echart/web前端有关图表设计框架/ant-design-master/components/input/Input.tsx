import * as React from 'react';
import * as PropTypes from 'prop-types';
import classNames from 'classnames';
import omit from 'omit.js';
import { polyfill } from 'react-lifecycles-compat';
import Group from './Group';
import Search from './Search';
import TextArea from './TextArea';
import { ConfigConsumer, ConfigConsumerProps } from '../config-provider';
import Password from './Password';
import Icon from '../icon';
import { Omit, tuple } from '../_util/type';

function fixControlledValue<T>(value: T) {
  if (typeof value === 'undefined' || value === null) {
    return '';
  }
  return value;
}

const InputSizes = tuple('small', 'default', 'large');

export interface InputProps
  extends Omit<React.InputHTMLAttributes<HTMLInputElement>, 'size' | 'prefix'> {
  prefixCls?: string;
  size?: (typeof InputSizes)[number];
  onPressEnter?: React.KeyboardEventHandler<HTMLInputElement>;
  addonBefore?: React.ReactNode;
  addonAfter?: React.ReactNode;
  prefix?: React.ReactNode;
  suffix?: React.ReactNode;
  allowClear?: boolean;
}

class Input extends React.Component<InputProps, any> {
  static Group: typeof Group;
  static Search: typeof Search;
  static TextArea: typeof TextArea;
  static Password: typeof Password;

  static defaultProps = {
    type: 'text',
    disabled: false,
  };

  static propTypes = {
    type: PropTypes.string,
    id: PropTypes.string,
    size: PropTypes.oneOf(InputSizes),
    maxLength: PropTypes.number,
    disabled: PropTypes.bool,
    value: PropTypes.any,
    defaultValue: PropTypes.any,
    className: PropTypes.string,
    addonBefore: PropTypes.node,
    addonAfter: PropTypes.node,
    prefixCls: PropTypes.string,
    onPressEnter: PropTypes.func,
    onKeyDown: PropTypes.func,
    onKeyUp: PropTypes.func,
    onFocus: PropTypes.func,
    onBlur: PropTypes.func,
    prefix: PropTypes.node,
    suffix: PropTypes.node,
    allowClear: PropTypes.bool,
  };

  static getDerivedStateFromProps(nextProps: InputProps) {
    if ('value' in nextProps) {
      return {
        value: nextProps.value,
      };
    }
    return null;
  }

  input: HTMLInputElement;

  constructor(props: InputProps) {
    super(props);
    const value = typeof props.value === 'undefined' ? props.defaultValue : props.value;
    this.state = {
      value,
    };
  }

  handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
    const { onPressEnter, onKeyDown } = this.props;
    if (e.keyCode === 13 && onPressEnter) {
      onPressEnter(e);
    }
    if (onKeyDown) {
      onKeyDown(e);
    }
  };

  focus() {
    this.input.focus();
  }

  blur() {
    this.input.blur();
  }

  select() {
    this.input.select();
  }

  getInputClassName(prefixCls: string) {
    const { size, disabled } = this.props;
    return classNames(prefixCls, {
      [`${prefixCls}-sm`]: size === 'small',
      [`${prefixCls}-lg`]: size === 'large',
      [`${prefixCls}-disabled`]: disabled,
    });
  }

  saveInput = (node: HTMLInputElement) => {
    this.input = node;
  };

  setValue(
    value: string,
    e: React.ChangeEvent<HTMLInputElement> | React.MouseEvent<HTMLElement, MouseEvent>,
  ) {
    if (!('value' in this.props)) {
      this.setState({ value });
    }
    const { onChange } = this.props;
    if (onChange) {
      let event = e;
      if (e.type === 'click') {
        // click clear icon
        event = Object.create(e);
        event.target = this.input;
        event.currentTarget = this.input;
        const originalInputValue = this.input.value;
        // change input value cause e.target.value should be '' when clear input
        this.input.value = '';
        onChange(event as React.ChangeEvent<HTMLInputElement>);
        // reset input value
        this.input.value = originalInputValue;
        return;
      }
      onChange(event as React.ChangeEvent<HTMLInputElement>);
    }
  }

  handleReset = (e: React.MouseEvent<HTMLElement, MouseEvent>) => {
    this.setValue('', e);
  };

  handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    this.setValue(e.target.value, e);
  };

  renderClearIcon(prefixCls: string) {
    const { allowClear } = this.props;
    const { value } = this.state;
    if (!allowClear || value === undefined || value === '') {
      return null;
    }
    return (
      <Icon
        type="close-circle"
        theme="filled"
        onClick={this.handleReset}
        className={`${prefixCls}-clear-icon`}
        role="button"
      />
    );
  }

  renderSuffix(prefixCls: string) {
    const { suffix, allowClear } = this.props;
    if (suffix || allowClear) {
      return (
        <span className={`${prefixCls}-suffix`}>
          {this.renderClearIcon(prefixCls)}
          {suffix}
        </span>
      );
    }
    return null;
  }

  renderLabeledInput(prefixCls: string, children: React.ReactElement<any>) {
    const { addonBefore, addonAfter, style, size, className } = this.props;
    // Not wrap when there is not addons
    if (!addonBefore && !addonAfter) {
      return children;
    }

    const wrapperClassName = `${prefixCls}-group`;
    const addonClassName = `${wrapperClassName}-addon`;
    const addonBeforeNode = addonBefore ? (
      <span className={addonClassName}>{addonBefore}</span>
    ) : null;
    const addonAfterNode = addonAfter ? <span className={addonClassName}>{addonAfter}</span> : null;

    const mergedWrapperClassName = classNames(`${prefixCls}-wrapper`, {
      [wrapperClassName]: addonBefore || addonAfter,
    });

    const mergedGroupClassName = classNames(className, `${prefixCls}-group-wrapper`, {
      [`${prefixCls}-group-wrapper-sm`]: size === 'small',
      [`${prefixCls}-group-wrapper-lg`]: size === 'large',
    });

    // Need another wrapper for changing display:table to display:inline-block
    // and put style prop in wrapper
    return (
      <span className={mergedGroupClassName} style={style}>
        <span className={mergedWrapperClassName}>
          {addonBeforeNode}
          {React.cloneElement(children, { style: null })}
          {addonAfterNode}
        </span>
      </span>
    );
  }

  renderLabeledIcon(prefixCls: string, children: React.ReactElement<any>) {
    const { props } = this;
    const suffix = this.renderSuffix(prefixCls);

    if (!('prefix' in props) && !suffix) {
      return children;
    }

    const prefix = props.prefix ? (
      <span className={`${prefixCls}-prefix`}>{props.prefix}</span>
    ) : null;

    const affixWrapperCls = classNames(props.className, `${prefixCls}-affix-wrapper`, {
      [`${prefixCls}-affix-wrapper-sm`]: props.size === 'small',
      [`${prefixCls}-affix-wrapper-lg`]: props.size === 'large',
    });
    return (
      <span className={affixWrapperCls} style={props.style}>
        {prefix}
        {React.cloneElement(children, {
          style: null,
          className: this.getInputClassName(prefixCls),
        })}
        {suffix}
      </span>
    );
  }

  renderInput(prefixCls: string) {
    const { className, addonBefore, addonAfter } = this.props;
    const { value } = this.state;
    // Fix https://fb.me/react-unknown-prop
    const otherProps = omit(this.props, [
      'prefixCls',
      'onPressEnter',
      'addonBefore',
      'addonAfter',
      'prefix',
      'suffix',
      'allowClear',
      // Input elements must be either controlled or uncontrolled,
      // specify either the value prop, or the defaultValue prop, but not both.
      'defaultValue',
    ]);

    return this.renderLabeledIcon(
      prefixCls,
      <input
        {...otherProps}
        value={fixControlledValue(value)}
        onChange={this.handleChange}
        className={classNames(this.getInputClassName(prefixCls), {
          [className!]: className && !addonBefore && !addonAfter,
        })}
        onKeyDown={this.handleKeyDown}
        ref={this.saveInput}
      />,
    );
  }

  renderComponent = ({ getPrefixCls }: ConfigConsumerProps) => {
    const { prefixCls: customizePrefixCls } = this.props;
    const prefixCls = getPrefixCls('input', customizePrefixCls);
    return this.renderLabeledInput(prefixCls, this.renderInput(prefixCls));
  };

  render() {
    return <ConfigConsumer>{this.renderComponent}</ConfigConsumer>;
  }
}

polyfill(Input);

export default Input;
