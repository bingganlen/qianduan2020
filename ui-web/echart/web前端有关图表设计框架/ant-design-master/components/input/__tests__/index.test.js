import React from 'react';
import { mount } from 'enzyme';
import Input from '..';
import Form from '../../form';
import focusTest from '../../../tests/shared/focusTest';

const { TextArea } = Input;

describe('Input', () => {
  focusTest(Input);

  it('should support maxLength', () => {
    const wrapper = mount(<Input maxLength={3} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('select()', () => {
    const wrapper = mount(<Input />);
    wrapper.instance().select();
  });
});

focusTest(TextArea);

describe('TextArea', () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  it('should auto calculate height according to content length', () => {
    const wrapper = mount(<TextArea value="" readOnly autosize />);
    const mockFunc = jest.spyOn(wrapper.instance(), 'resizeTextarea');
    wrapper.setProps({ value: '1111\n2222\n3333' });
    jest.runAllTimers();
    expect(mockFunc).toHaveBeenCalledTimes(1);
    wrapper.setProps({ value: '1111' });
    jest.runAllTimers();
    expect(mockFunc).toHaveBeenCalledTimes(2);
  });

  it('should support disabled', () => {
    const wrapper = mount(<TextArea disabled />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should support maxLength', () => {
    const wrapper = mount(<TextArea maxLength={10} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('As Form Control', () => {
  it('should be reset when wrapped in form.getFieldDecorator without initialValue', () => {
    class Demo extends React.Component {
      reset = () => {
        const { form } = this.props;
        form.resetFields();
      };

      render() {
        const {
          form: { getFieldDecorator },
        } = this.props;
        return (
          <Form>
            <Form.Item>{getFieldDecorator('input')(<Input />)}</Form.Item>
            <Form.Item>{getFieldDecorator('textarea')(<Input.TextArea />)}</Form.Item>
            <button type="button" onClick={this.reset}>
              reset
            </button>
          </Form>
        );
      }
    }
    const DemoForm = Form.create()(Demo);
    const wrapper = mount(<DemoForm />);
    wrapper.find('input').simulate('change', { target: { value: '111' } });
    wrapper.find('textarea').simulate('change', { target: { value: '222' } });
    expect(wrapper.find('input').prop('value')).toBe('111');
    expect(wrapper.find('textarea').prop('value')).toBe('222');
    wrapper.find('button').simulate('click');
    expect(wrapper.find('input').prop('value')).toBe('');
    expect(wrapper.find('textarea').prop('value')).toBe('');
  });
});

describe('Input.Search', () => {
  it('should support suffix', () => {
    const wrapper = mount(<Input.Search suffix="suffix" />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Input.Password', () => {
  it('should change type when click', () => {
    const wrapper = mount(<Input.Password />);
    wrapper.find('input').simulate('change', { target: { value: '111' } });
    expect(wrapper).toMatchSnapshot();
    wrapper
      .find('.ant-input-password-icon')
      .at(0)
      .simulate('click');
    expect(wrapper).toMatchSnapshot();
    wrapper
      .find('.ant-input-password-icon')
      .at(0)
      .simulate('click');
    expect(wrapper).toMatchSnapshot();
  });

  it('visibilityToggle should work', () => {
    const wrapper = mount(<Input.Password visibilityToggle={false} />);
    expect(wrapper.find('.anticon-eye').length).toBe(0);
    wrapper.setProps({ visibilityToggle: true });
    expect(wrapper.find('.anticon-eye').length).toBe(1);
  });
});

describe('Input allowClear', () => {
  it('should change type when click', () => {
    const wrapper = mount(<Input allowClear />);
    wrapper.find('input').simulate('change', { target: { value: '111' } });
    expect(wrapper.find('input').getDOMNode().value).toEqual('111');
    expect(wrapper).toMatchSnapshot();
    wrapper
      .find('.ant-input-clear-icon')
      .at(0)
      .simulate('click');
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('input').getDOMNode().value).toEqual('');
  });

  it('should trigger event correctly', () => {
    let argumentEventObject;
    let argumentEventObjectValue;
    const onChange = e => {
      argumentEventObject = e;
      argumentEventObjectValue = e.target.value;
    };
    const wrapper = mount(<Input allowClear defaultValue="111" onChange={onChange} />);
    wrapper
      .find('.ant-input-clear-icon')
      .at(0)
      .simulate('click');
    expect(argumentEventObject.type).toBe('click');
    expect(argumentEventObjectValue).toBe('');
    expect(
      wrapper
        .find('input')
        .at(0)
        .getDOMNode().value,
    ).toBe('');
  });

  it('should trigger event correctly on controlled mode', () => {
    let argumentEventObject;
    let argumentEventObjectValue;
    const onChange = e => {
      argumentEventObject = e;
      argumentEventObjectValue = e.target.value;
    };
    const wrapper = mount(<Input allowClear value="111" onChange={onChange} />);
    wrapper
      .find('.ant-input-clear-icon')
      .at(0)
      .simulate('click');
    expect(argumentEventObject.type).toBe('click');
    expect(argumentEventObjectValue).toBe('');
    expect(
      wrapper
        .find('input')
        .at(0)
        .getDOMNode().value,
    ).toBe('111');
  });
});
