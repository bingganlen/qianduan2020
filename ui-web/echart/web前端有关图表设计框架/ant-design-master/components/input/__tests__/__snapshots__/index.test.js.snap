// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input allowClear should change type when click 1`] = `
<Input
  allowClear={true}
  disabled={false}
  type="text"
>
  <Consumer>
    <span
      className="ant-input-affix-wrapper"
    >
      <input
        className="ant-input"
        disabled={false}
        onChange={[Function]}
        onKeyDown={[Function]}
        style={null}
        type="text"
        value="111"
      />
      <span
        className="ant-input-suffix"
      >
        <Icon
          className="ant-input-clear-icon"
          onClick={[Function]}
          role="button"
          theme="filled"
          type="close-circle"
        >
          <LocaleReceiver
            componentName="Icon"
          >
            <i
              aria-label="icon: close-circle"
              className="anticon anticon-close-circle ant-input-clear-icon"
              onClick={[Function]}
              role="button"
              tabIndex={-1}
            >
              <IconReact
                className=""
                type="close-circle-fill"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="close-circle"
                  fill="currentColor"
                  height="1em"
                  key="svg-close-circle"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 0 1-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                    key="svg-close-circle-svg-0"
                  />
                </svg>
              </IconReact>
            </i>
          </LocaleReceiver>
        </Icon>
      </span>
    </span>
  </Consumer>
</Input>
`;

exports[`Input allowClear should change type when click 2`] = `
<Input
  allowClear={true}
  disabled={false}
  type="text"
>
  <Consumer>
    <span
      className="ant-input-affix-wrapper"
    >
      <input
        className="ant-input"
        disabled={false}
        onChange={[Function]}
        onKeyDown={[Function]}
        style={null}
        type="text"
        value=""
      />
      <span
        className="ant-input-suffix"
      />
    </span>
  </Consumer>
</Input>
`;

exports[`Input should support maxLength 1`] = `
<Input
  disabled={false}
  maxLength={3}
  type="text"
>
  <Consumer>
    <input
      className="ant-input"
      disabled={false}
      maxLength={3}
      onChange={[Function]}
      onKeyDown={[Function]}
      type="text"
      value=""
    />
  </Consumer>
</Input>
`;

exports[`Input.Password should change type when click 1`] = `
<Password
  action="click"
  inputPrefixCls="ant-input"
  prefixCls="ant-input-password"
  visibilityToggle={true}
>
  <Input
    action="click"
    className="ant-input-password"
    disabled={false}
    prefixCls="ant-input"
    suffix={
      <Icon
        className="ant-input-password-icon"
        onClick={[Function]}
        type="eye"
      />
    }
    type="password"
  >
    <Consumer>
      <span
        className="ant-input-password ant-input-affix-wrapper"
      >
        <input
          action="click"
          className="ant-input"
          disabled={false}
          onChange={[Function]}
          onKeyDown={[Function]}
          style={null}
          type="password"
          value="111"
        />
        <span
          className="ant-input-suffix"
        >
          <Icon
            className="ant-input-password-icon"
            key="passwordIcon"
            onClick={[Function]}
            type="eye"
          >
            <LocaleReceiver
              componentName="Icon"
            >
              <i
                aria-label="icon: eye"
                className="anticon anticon-eye ant-input-password-icon"
                onClick={[Function]}
                tabIndex={-1}
              >
                <IconReact
                  className=""
                  type="eye-o"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="eye"
                    fill="currentColor"
                    height="1em"
                    key="svg-eye"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 0 0 0 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                      key="svg-eye-svg-0"
                    />
                  </svg>
                </IconReact>
              </i>
            </LocaleReceiver>
          </Icon>
        </span>
      </span>
    </Consumer>
  </Input>
</Password>
`;

exports[`Input.Password should change type when click 2`] = `
<Password
  action="click"
  inputPrefixCls="ant-input"
  prefixCls="ant-input-password"
  visibilityToggle={true}
>
  <Input
    action="click"
    className="ant-input-password"
    disabled={false}
    prefixCls="ant-input"
    suffix={
      <Icon
        className="ant-input-password-icon"
        onClick={[Function]}
        type="eye-invisible"
      />
    }
    type="text"
  >
    <Consumer>
      <span
        className="ant-input-password ant-input-affix-wrapper"
      >
        <input
          action="click"
          className="ant-input"
          disabled={false}
          onChange={[Function]}
          onKeyDown={[Function]}
          style={null}
          type="text"
          value="111"
        />
        <span
          className="ant-input-suffix"
        >
          <Icon
            className="ant-input-password-icon"
            key="passwordIcon"
            onClick={[Function]}
            type="eye-invisible"
          >
            <LocaleReceiver
              componentName="Icon"
            >
              <i
                aria-label="icon: eye-invisible"
                className="anticon anticon-eye-invisible ant-input-password-icon"
                onClick={[Function]}
                tabIndex={-1}
              >
                <IconReact
                  className=""
                  type="eye-invisible-o"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="eye-invisible"
                    fill="currentColor"
                    height="1em"
                    key="svg-eye-invisible"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 0 0 0-51.5zm-63.57-320.64L836 122.88a8 8 0 0 0-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 0 0 0 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 0 0 0 11.31L155.17 889a8 8 0 0 0 11.31 0l712.15-712.12a8 8 0 0 0 0-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 0 0-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 0 1 146.2-106.69L401.31 546.2A112 112 0 0 1 396 512z"
                      key="svg-eye-invisible-svg-0"
                    />
                    <path
                      d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 0 0 227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 0 1-112 112z"
                      key="svg-eye-invisible-svg-1"
                    />
                  </svg>
                </IconReact>
              </i>
            </LocaleReceiver>
          </Icon>
        </span>
      </span>
    </Consumer>
  </Input>
</Password>
`;

exports[`Input.Password should change type when click 3`] = `
<Password
  action="click"
  inputPrefixCls="ant-input"
  prefixCls="ant-input-password"
  visibilityToggle={true}
>
  <Input
    action="click"
    className="ant-input-password"
    disabled={false}
    prefixCls="ant-input"
    suffix={
      <Icon
        className="ant-input-password-icon"
        onClick={[Function]}
        type="eye"
      />
    }
    type="password"
  >
    <Consumer>
      <span
        className="ant-input-password ant-input-affix-wrapper"
      >
        <input
          action="click"
          className="ant-input"
          disabled={false}
          onChange={[Function]}
          onKeyDown={[Function]}
          style={null}
          type="password"
          value="111"
        />
        <span
          className="ant-input-suffix"
        >
          <Icon
            className="ant-input-password-icon"
            key="passwordIcon"
            onClick={[Function]}
            type="eye"
          >
            <LocaleReceiver
              componentName="Icon"
            >
              <i
                aria-label="icon: eye"
                className="anticon anticon-eye ant-input-password-icon"
                onClick={[Function]}
                tabIndex={-1}
              >
                <IconReact
                  className=""
                  type="eye-o"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="eye"
                    fill="currentColor"
                    height="1em"
                    key="svg-eye"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 0 0 0 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                      key="svg-eye-svg-0"
                    />
                  </svg>
                </IconReact>
              </i>
            </LocaleReceiver>
          </Icon>
        </span>
      </span>
    </Consumer>
  </Input>
</Password>
`;

exports[`Input.Search should support suffix 1`] = `
<Search
  enterButton={false}
  suffix="suffix"
>
  <Consumer>
    <Input
      addonAfter={null}
      className="ant-input-search"
      disabled={false}
      onPressEnter={[Function]}
      prefixCls="ant-input"
      suffix={
        Array [
          "suffix",
          <Icon
            className="ant-input-search-icon"
            onClick={[Function]}
            type="search"
          />,
        ]
      }
      type="text"
    >
      <Consumer>
        <span
          className="ant-input-search ant-input-affix-wrapper"
        >
          <input
            className="ant-input"
            disabled={false}
            onChange={[Function]}
            onKeyDown={[Function]}
            style={null}
            type="text"
            value=""
          />
          <span
            className="ant-input-suffix"
          >
            suffix
            <Icon
              className="ant-input-search-icon"
              key="searchIcon"
              onClick={[Function]}
              type="search"
            >
              <LocaleReceiver
                componentName="Icon"
              >
                <i
                  aria-label="icon: search"
                  className="anticon anticon-search ant-input-search-icon"
                  onClick={[Function]}
                  tabIndex={-1}
                >
                  <IconReact
                    className=""
                    type="search-o"
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-icon="search"
                      fill="currentColor"
                      height="1em"
                      key="svg-search"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                        key="svg-search-svg-0"
                      />
                    </svg>
                  </IconReact>
                </i>
              </LocaleReceiver>
            </Icon>
          </span>
        </span>
      </Consumer>
    </Input>
  </Consumer>
</Search>
`;

exports[`TextArea should support disabled 1`] = `
<TextArea
  disabled={true}
>
  <Consumer>
    <textarea
      className="ant-input ant-input-disabled"
      disabled={true}
      onChange={[Function]}
      onKeyDown={[Function]}
      style={Object {}}
    />
  </Consumer>
</TextArea>
`;

exports[`TextArea should support maxLength 1`] = `
<TextArea
  maxLength={10}
>
  <Consumer>
    <textarea
      className="ant-input"
      maxLength={10}
      onChange={[Function]}
      onKeyDown={[Function]}
      style={Object {}}
    />
  </Consumer>
</TextArea>
`;
